### PR TITLE
feat: implement `access_token` auth method

### PR DESCRIPTION
**Pull request by 0x416e746f6e**
subj.

this PR is a part 1 for issue https://github.com/ovh/terraform-provider-ovh/issues/660

cc: @amstuta

### CODE REVIEW

**Comment by deathiop** [comment]
Hello,

Thank you for this implementation!
As per our [contributing guide](https://github.com/ovh/go-ovh/blob/master/CONTRIBUTING.md), we require commits to be signed-off. Could you please amend yours? (`git commit --amend --no-edit -s` would do the trick)

**Comment by 0x416e746f6e** [comment]
> Hello,
> 
> Thank you for this implementation! As per our [contributing guide](https://github.com/ovh/go-ovh/blob/master/CONTRIBUTING.md), we require commits to be signed-off. Could you please amend yours? (`git commit --amend --no-edit -s` would do the trick)

hi @deathiop, but my commit is already signed:

image

or am I missing something?

**Comment by 0x416e746f6e** [comment]
ah, ok.  I see what you mean.

fixed!

File: ovh/configuration.go [hunk 1]
```diff
@@ -125,19 +129,33 @@ func (c *Client) loadConfig(endpointName string) error {
...
+	if c.AccessToken != "" {
+		configuredAuthMethods = append(configuredAuthMethods, "access_token")
+	}
+
+	if len(configuredAuthMethods) > 1 {
+		return fmt.Errorf("can't use multiple authentication methods: %s", strings.Join(configuredAuthMethods, ", "))
+	}
+	if len(configuredAuthMethods) == 0 {
+		return errors.New(
+			"missing authentication information, you need to provide at least an application_key/application_secret, or client_id/client_secret, or access_token",
```
Comment (amstuta) [inline ovh/configuration.go:148]:
```suggestion
			"missing authentication information, you need to provide one of the following: application_key/application_secret, client_id/client_secret, or access_token",
```

**Review by amstuta — COMMENTED** [review]
Thanks for your contribution :) 

Could you also add a section in https://github.com/ovh/go-ovh?tab=readme-ov-file#configuration to describe this new auth method ?

**Comment by 0x416e746f6e** [comment]
> Could you also add a section in https://github.com/ovh/go-ovh?tab=readme-ov-file#configuration to describe this new auth method ?

@amstuta please have a look.

File: ovh/configuration_test.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func TestConfigFromNonExistingFile(t *testing.T) {
 
 	client := Client{}
 	err := client.loadConfig("ovh-eu")
-	td.CmpString(t, err, `missing authentication information, you need to provide at least an application_key/application_secret or a client_id/client_secret`)
+	td.CmpString(t, err, `missing authentication information, you need to provide at least an application_key/application_secret, or client_id/client_secret, or access_token`)
```
Comment (amstuta) [inline ovh/configuration_test.go:67]: This needs to be updated with the new error message.

File: README.md [hunk 3]
```diff
@@ -99,6 +99,21 @@ Depending on the API you want to use, you may set the ``endpoint`` to:
 This lookup mechanism makes it easy to overload credentials for a specific
 project or user.
 
+### Access Token
+
+This authentication method is useful when short-lived credentials are necessary.
+E.g. oauth2 [plugin](https://github.com/puppetlabs/vault-plugin-secrets-oauthapp)
+for hashicorp vault can request an access token that would be used by OVH
+terraform provider. Although this token, requested via data-source, would end up
+stored in the terraform state-file, that would pose less risk since the token
```
Comment (deathiop) [inline README.md:108]:
```suggestion
stored in the Terraform state file, that would pose less risk since the token
```

**Pull request merged**